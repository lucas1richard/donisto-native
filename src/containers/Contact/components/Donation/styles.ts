import { StyleSheet } from 'react-native';
import theme from 'theme/iftheme';

const styles = StyleSheet.create({
  wrapper: {
    width: theme.fs * 12,
    height: theme.fs * 12,
    padding: theme.fs
  },
  innerWrapper: {
    height: '100%',
    width: '100%',
    backgroundColor: 'red',
    alignItems: 'center',
    justifyContent: 'center'
  }
});

export default styles;