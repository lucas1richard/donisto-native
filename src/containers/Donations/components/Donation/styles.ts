import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  marginWrapper: {
    width: '50%',
    height: 240,
    padding: 7.5
  },
  outerWrapper: {
    height: '100%',
    borderRadius: 10,
    elevation: 3,
    shadowColor: '#555',
    shadowOpacity: 0.3,
    shadowOffset: {
      width: 2,
      height: 4
    },
    shadowRadius: 6,
    backgroundColor: '#fff',
    justifyContent: 'space-between',
  },
  imageWrapper: {
    overflow: 'hidden',
    height: '67%',
    backgroundColor: '#aaa',
    width: '100%',
    borderTopLeftRadius: 10,
    borderTopRightRadius: 10
  },
  image: {
    height: '100%',
    width: '100%'
  }
});

export default styles;
