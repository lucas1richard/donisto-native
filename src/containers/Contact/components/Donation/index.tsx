import * as React from 'react';
import { View } from 'react-native';
import Touchable from 'components/Touchable';
import H2 from 'components/H2';
import styles from './styles';

interface IDonationProps {
  title: string;
}

class Donation extends React.Component<IDonationProps> {
  render() {
    const { title } = this.props;
    return (
      <View style={styles.wrapper}>
        <Touchable>
          <View style={styles.innerWrapper}>
            <H2 color="light">{title}</H2>
          </View>
        </Touchable>
      </View>
    );
  }
}

export default Donation;
