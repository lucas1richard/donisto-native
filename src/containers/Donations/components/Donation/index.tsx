import React from 'react';
import { View, Image } from 'react-native';
import Txt from 'components/Txt';
import styles from './styles';
import Touchable from 'components/Touchable';

class Donation extends React.Component<any> {
  render() {
    const { img, onPress } = this.props;
    return (
      <View style={styles.marginWrapper}>
        <Touchable onPress={onPress}>
          <View style={styles.outerWrapper}>
            <View style={styles.imageWrapper}>
              <Image source={img} style={styles.image} />
            </View>
            <View style={{ padding: 10 }}>
              <Txt style={{ fontSize: 13 }}>
                School for Kids
              </Txt>
              <Txt style={{ fontSize: 13 }}>
                Pre-K to Junior High
              </Txt>
              <Txt style={{ fontSize: 13, color: '#aaa' }}>
                on Dec 14, 2018
              </Txt>
            </View>
          </View>
        </Touchable>
      </View>
    );
  }
}

export default Donation;