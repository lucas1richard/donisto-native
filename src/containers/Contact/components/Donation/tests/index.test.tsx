import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Donation, { IDonationProps } from '..';

describe('<Donation />', () => {
  const renderComponent = (props: IDonationProps) => toJson(shallow(<Donation {...props} />));
  it('renders with just text', () => {
    expect(renderComponent({ title: 'mockTitle' })).toMatchSnapshot();
  });
});