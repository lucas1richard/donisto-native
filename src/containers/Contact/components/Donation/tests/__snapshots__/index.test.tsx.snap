// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Donation /> renders with just text 1`] = `
<Component
  style={
    Object {
      "height": 240,
      "padding": 20,
      "width": 240,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <H2
        color="light"
      >
        mockTitle
      </H2>
    </Component>
  </TouchableOpacity>
</Component>
`;
